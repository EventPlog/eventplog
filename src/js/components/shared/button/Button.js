import React from 'react'
import styled, { css } from 'styled-components'
import colors from '../../../styles/theme/colors'
import { NavLink } from 'react-router-dom';
import { media } from 'js/styles/mixins'

const btnSizes = {
  medium: {fontSize: '1.2em'},
}
const commonStyles = css`
  border: 1px solid var(--activeLink);
  color: var(--activeLink);
  background: transparent;
  padding: 10px 30px;
  border-radius: 5px;
  text-transform: uppercase;
  font-size: 0.8rem;
  font-weight: 600;
  letter-spacing: 1px;
  
  ${
    media.phone`
      padding: 0.8rem;
    `
  }
  &:hover {
    background: var(--activeLink);
    color: ${colors.white};
  }
`

const inverted = css`
  background: var(--activeLink);
  color: ${ colors.white} !important;
  
  &:hover {
    background: ${ colors.white };
    color: var(--activeLink) !important;
  }
`

const StyledBtn = styled.button`
  ${ commonStyles }
`

const StyledLink = styled(NavLink)`
  ${ commonStyles }
  
  &:hover {
    color: ${colors.white} !important;
  }
`;

const InvertedBtn = (Component, props) => (
  props.inverted
  ? Component.extend`
      ${ inverted }
    `
  : Component
)

const Button = function(props) {
  const Component = InvertedBtn(StyledBtn, props)
  return <Component {...props} />
}

Button.Link = function(props) {
  const Component = InvertedBtn(StyledLink, props)
  return <Component {...props} />
}


export default Button
