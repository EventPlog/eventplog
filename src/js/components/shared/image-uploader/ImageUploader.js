import React from 'react'
import { Icon } from 'semantic-ui-react'
import styled from 'styled-components'


// ======= INTERNAL =========
import Button from 'js/components/shared/button'

const StyledLoader = styled.div`
`

const ImageUploader = ({
  image,
  imageInputRef,
  handleImageChange,
  showImageSelectOptions,
  saveImage,
  cancelChange,
}) => {
  return (
    <StyledLoader>
      <input ref={imageInputRef}
             onChange={handleImageChange}
             className="hidden"
             id="upload-img"
             type="file"
             name="image"
             accept="image/*" />
      {!image &&
        <Button className="btn-right upload-btn"
                onClick={showImageSelectOptions}>
          <Icon className="image" />
        </Button>}
      {image &&
        <span>
          <Button className="btn-right save-btn"
                  onClick={saveImage}>
            Save
          </Button>
          <Button className="btn-right cancel-btn"
                  onClick={cancelChange}>
            Cancel
          </Button>
        </span>
      }
    </StyledLoader>
  )
}

export default ImageUploader