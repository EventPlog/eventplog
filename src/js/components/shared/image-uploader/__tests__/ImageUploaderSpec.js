import React from 'react';
import ImageUploader from '../ImageUploader';
import Button from 'js/components/shared/button'
import { shallow } from 'enzyme';

describe('ImageUploader', () => {

  it('should render correctly', () => {
    const wrapper = shallow( <ImageUploader /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(Button).length).toEqual(1)
  })
});