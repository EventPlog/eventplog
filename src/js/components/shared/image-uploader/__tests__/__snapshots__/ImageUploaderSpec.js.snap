// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploader should render correctly 1`] = `
<styled.div>
  <input
    accept="image/*"
    className="hidden"
    id="upload-img"
    name="image"
    type="file"
  />
  <Button
    className="btn-right upload-btn"
  >
    <Icon
      as="i"
      className="image"
    />
  </Button>
</styled.div>
`;
